tools/slicec-cs/src/generators/exception_generator.rs

```diff
@@ -44,118 +56,144 @@ pub fn generate_exception(exception_def: &Exception) -> CodeBlock {
-        .add_parameter("ref SliceEncoder", "encoder", None, None)
-        .set_body(
-            format!(
-                "\
-encoder.StartSlice(SliceTypeId);
-{encode_fields}
-encoder.EndSlice(lastSlice: {is_last_slice});
-{encode_base}",
-                encode_fields = encode_fields(fields, Encoding::Slice1),
-                is_last_slice = !has_base,
-                encode_base = if has_base { "base.EncodeCore(ref encoder);" } else { "" },
-            )
-            .into(),
-        )
-        .build()
+    builder.build()
 }
 
-fn one_shot_constructor(exception_def: &Exception) -> CodeBlock {
-    let exception_name = exception_def.escape_identifier();
+fn constructor(
+    escaped_name: &str,
+    access: &str,
+    summary_comment: String,
+    namespace: &str,
+    fields: &[&Field],
+    base_fields: &[&Field],
+) -> CodeBlock {
+    let mut code = CodeBlock::default();
 
-    let namespace = &exception_def.namespace();
+    let mut builder = FunctionBuilder::new(access, "", escaped_name, FunctionType::BlockBody);
 
-    let all_fields = exception_def.all_fields();
-
-    let base_parameters = if let Some(base) = exception_def.base_exception() {
-        base.all_fields().iter().map(|m| m.parameter_name()).collect::<Vec<_>>()
-    } else {
-        vec![]
-    };
-
-    let mut ctor_builder = FunctionBuilder::new("public", "", &exception_name, FunctionType::BlockBody);
-
-    if all_fields.iter().any(|f| f.is_required()) {
-        ctor_builder.add_attribute("global::System.Diagnostics.CodeAnalysis.SetsRequiredMembers");
+    if fields.iter().any(|f| f.is_required()) || base_fields.iter().any(|f| f.is_required()) {
+        builder.add_attribute("global::System.Diagnostics.CodeAnalysis.SetsRequiredMembers");
     }
 
-    ctor_builder.add_comment(
-        "summary",
-        format!(r#"Constructs a new instance of <see cref="{}" />."#, &exception_name),
-    );
+    builder.add_comment("summary", summary_comment);
+
+    builder.add_base_parameters(&base_fields.iter().map(|m| m.parameter_name()).collect::<Vec<String>>());
 
-    for field in &all_fields {
-        ctor_builder.add_parameter(
-            &field.data_type().field_type_string(namespace),
-            field.parameter_name().as_str(),
+    for field in base_fields.iter().chain(fields.iter()) {
+        builder.add_parameter(
+            &field.data_type.field_type_string(namespace),
+            &field.parameter_name(),
             None,
             field.formatted_doc_comment_summary(),
         );
     }
-    ctor_builder.add_base_parameters(&base_parameters);
 
-    // ctor impl
-    let mut ctor_body = CodeBlock::default();
-    for field in exception_def.fields() {
-        let field_name = field.field_name();
-        let parameter_name = field.parameter_name();
+    builder.set_body({
+        let mut code = CodeBlock::default();
+        for field in fields {
+            writeln!(code, "this.{} = {};", field.field_name(), field.parameter_name(),);
```

Comment:
Looks like a bug introduced by rustfmt. Obviously there shouldn't be a trailing comma here.
```suggestion
            writeln!(code, "this.{} = {};", field.field_name(), field.parameter_name());
```